packages/styled/types/base.d.ts

```diff
@@ -105,7 +105,7 @@ export interface CreateStyled {
     > = keyof React.ComponentProps<C>
   >(
     component: C,
-    options: FilteringStyledOptions<PropsOf<C>, ForwardedProps>
+    options: FilteringStyledOptions<React.ComponentProps<C>, ForwardedProps>
```

Comment:
this is not an equivalent, but actually I think the new version is more correct - we don't need to resolve defaultProps here (I don't care as much about resolving propTypes)

the even more correct version would use `React.ComponentPropsWithoutRef` but that is using conditional type + Exclude and Pick combo, so probably better to avoid it if we can, and the gain from using this over a simpler `React.ComponentProps` wouldnt be that big